src/main/scala/com/databricks/spark/avro/SchemaConverters.scala

```diff
@@ -90,8 +90,17 @@ object SchemaConverters {
             SchemaType(LongType, nullable = false)
           case Seq(t1, t2) if Set(t1, t2) == Set(FLOAT, DOUBLE) =>
             SchemaType(DoubleType, nullable = false)
-          case other => throw new IncompatibleSchemaException(
-            s"This mix of union types is not supported (see README): $other")
+          case _ =>
+            // Convert complex unions to struct types where field names are member0, member1, etc.
+            // This is consistent with the behavior when reading Parquet files.
```

Comment:
By "reading Parquet files", I guess you mean reading Avro records as Parquet messages using parquet-avro?